File: sdk/python/dogdata/resources/transactions.py
from __future__ import annotations

from typing import Any, Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from dogdata.client import DogData


class TransactionsResource:
    def __init__(self, client: DogData):
        self._client = client

    def list(self, page: int = 1, limit: int = 100) -> dict[str, Any]:
        return self._client.request("/api/dog-rune/transactions-kv", {
            "page": page,
            "limit": limit,
        })

    def get(self, txid: str) -> dict[str, Any]:
        return self._client.request("/api/dog-rune/transactions-kv", {
            "txid": txid,
        })

    def recent(self, limit: int = 20) -> list[dict[str, Any]]:
        result = self.list(page=1, limit=limit)
        return result.get("data", [])

    def by_address(self, address: str, page: int = 1, limit: int = 100) -> dict[str, Any]:
        return self._client.request("/api/dog-rune/transactions-kv", {
            "address": address,
            "page": page,
            "limit": limit,
        })

File: sdk/typescript/src/resources/transactions.ts
import type { DogData } from '../client.js';
import type { PaginationParams, PaginatedResponse, Transaction } from '../types.js';

export class TransactionsResource {
  constructor(private client: DogData) {}

  async list(params?: PaginationParams): Promise<PaginatedResponse<Transaction>> {
    return this.client.request<PaginatedResponse<Transaction>>('/api/dog-rune/transactions-kv', {
      page: params?.page,
      limit: params?.limit,
    });
  }

  async get(txid: string): Promise<Transaction> {
    return this.client.request<Transaction>('/api/dog-rune/transactions-kv', {
      txid,
    });
  }

  async recent(limit: number = 20): Promise<Transaction[]> {
    const result = await this.list({ page: 1, limit });
    return result.data;
  }

  async byAddress(address: string, params?: PaginationParams): Promise<PaginatedResponse<Transaction>> {
    return this.client.request<PaginatedResponse<Transaction>>('/api/dog-rune/transactions-kv', {
      address,
      page: params?.page,
      limit: params?.limit,
    });
  }
}
